From: Trae <trae@example.com>
Date: Thu, 12 Jun 2026 00:00:00 +0000
Subject: [PATCH] media: i2c: hi846: Accept 19.2MHz clock

The Xiaomi Pad 6 (Pipa) provides a 19.2MHz clock to the hi846 front
camera. The driver currently strictly requires 25MHz and warns/fails.
Allow 19.2MHz as a valid clock frequency.

---
 drivers/media/i2c/hi846.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/hi846.c b/drivers/media/i2c/hi846.c
index XXXXXXX..XXXXXXX 100644
--- a/drivers/media/i2c/hi846.c
+++ b/drivers/media/i2c/hi846.c
@@ -2059,9 +2059,9 @@ static int hi846_probe(struct i2c_client *client)
 				 hi846->clock);
 
 	mclk_freq = clk_get_rate(hi846->clock);
-if (mclk_freq != 25000000)
+if (mclk_freq != 25000000 && mclk_freq != 19200000)
 		dev_warn(&client->dev,
- "External clock freq should be 25000000, not %u.\n",
+ "External clock freq %u is non-standard (expected 19.2M or 25M).\n",
 			 mclk_freq);
 
 	for (i = 0; i < HI846_NUM_SUPPLIES; i++)
